### PR TITLE
Drop Python 3.8 support

### PR DESCRIPTION
**Pull request by hauntsaninja**
These are manual changes, we can run pyupgrade equivalents in a later PR

### CODE REVIEW

**Review by JelleZijlstra — APPROVED** [review]

File: src/black/__init__.py [hunk 1]
```diff
@@ -549,6 +549,7 @@ def main(  # noqa: C901
     """The uncompromising code formatter."""
     ctx.ensure_object(dict)
 
+    assert sys.version_info >= (3, 9), "Black requires Python 3.9+"
```
Comment (jtmedley) [inline src/black/__init__.py:552]: Why is this added? This wasn't needed in the `__init__.py` file for previous minimum versions. Python version requirements are handled by `requires-python` in [pyproject.toml](https://github.com/psf/black/blob/main/pyproject.toml), and that was already correctly changed in this commit.

Comment (hauntsaninja) [reply src/black/__init__.py:552]: It's harmless enough, Black is definitely broken on 3.8, I have dev environments lying around with random Python versions. Is it causing you some issue?
